agent/src/test/java/com/appland/appmap/record/RecorderTest.java

```diff
@@ -98,6 +104,78 @@ public void testUnwriteableOutputFile() throws IOException {
     assertTrue(exception.toString().indexOf("java.lang.RuntimeException: ") == 0);
   }
 
+  @Test
+  // for Files.move when: REPLACE_EXISTING, ATOMIC_MOVE
+  public void testWriteFileAcrossFilesystems() throws IOException {
+    if (!System.getProperty("os.name").equals("Linux")) {
+        //  /dev/shm exists only on Linux
+        return;
+    }
+    String sourceFilename = "/tmp/recordertest_file";
+    String targetFilename = "/dev/shm/recordertest_file";
+    File sourceFile = new File(sourceFilename);
+    File targetFile = new File(targetFilename);
+    Exception exception = null;
+    try {
+      // if the file exists createNewFile returns false
+      sourceFile.createNewFile();
+    } catch (IOException e) {
```

Comment:
Given that an `IOException` is unexpected here, I'd say it would be better to remove this `catch`, and just let the exception propagate (and cause the test to fail).

Reply:
Fixed, here and in `testCantOverwriteTargetFile`.